zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -241,6 +245,9 @@ private void assignAuthInfo(ServerCnxn cnxn, String clientId, Set<String> domain
     } else {
       // For other cases, add (x509:domainName) in authInfo
       domains.stream().forEach(d -> newAuthIds.add(new Id(getScheme(), d)));
+      if (domains.isEmpty() || X509AuthenticationConfig.getInstance().isStoreAuthedClientIdEnabled()) {
```

Comment:
neat : adding a comment for in which case domain list would be empty and in which can other things would be there would be nice.